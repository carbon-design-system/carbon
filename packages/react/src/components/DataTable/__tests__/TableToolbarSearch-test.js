/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import { TableToolbarSearch } from '../';

const prefix = 'cds';

describe('DataTable.TableToolbarSearch', () => {
  it('should render', () => {
    const wrapper = mount(
      <TableToolbarSearch
        className="custom-class"
        onChange={jest.fn()}
        id="custom-id"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should set an appropriate unique id', () => {
    const wrapper = mount(
      <TableToolbarSearch className="custom-class" onChange={jest.fn()} />
    );
    expect(typeof wrapper.find('Search').prop('id')).toBe('string');
  });
  it('should pass handleExpand as second parameter to onBlur/onFocus when provided', () => {
    const onBlur = jest.fn().mockImplementation((event, handleExpand) => {
      const { value } = event.target;
      if (!value) {
        handleExpand(event, false);
      }
    });
    const onFocus = jest.fn().mockImplementation((event, handleExpand) => {
      handleExpand(event, true);
    });
    const wrapper = mount(
      <TableToolbarSearch
        onChange={jest.fn()}
        onBlur={onBlur}
        onFocus={onFocus}
      />
    );
    wrapper.find('input').simulate('focus');
    expect(onFocus).toHaveBeenCalledTimes(1);
    expect(typeof onFocus.mock.calls[0][1]).toEqual('function');
    expect(wrapper.find('div').first().instance()).toHaveClass(
      `${prefix}--toolbar-search-container-active`
    );
    wrapper.find('input').simulate('blur', { target: { value: 'test' } });
    expect(onBlur).toHaveBeenCalledTimes(1);
    expect(typeof onBlur.mock.calls[0][1]).toEqual('function');

    // should continue to be expanded, because the input has a value.
    expect(wrapper.find('div').first().instance()).toHaveClass(
      `${prefix}--toolbar-search-container-active`
    );
  });

  it('should expand/contract as normal when no onBlur/onFocus provided', () => {
    const wrapper = mount(<TableToolbarSearch onChange={jest.fn()} />);
    wrapper.find('input').simulate('focus');
    expect(wrapper.find('div').first().instance()).toHaveClass(
      `${prefix}--toolbar-search-container-active`
    );
    wrapper.find('input').simulate('blur');
    expect(wrapper.find('div').first().instance()).not.toHaveClass(
      `${prefix}--toolbar-search-container-active`
    );
  });
});
