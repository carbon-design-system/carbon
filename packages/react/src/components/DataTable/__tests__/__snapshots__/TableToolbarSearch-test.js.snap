// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.TableToolbarSearch should render 1`] = `
<TableToolbarSearch
  className="custom-class"
  id="custom-id"
  onChange={[MockFunction]}
  persistent={false}
  tabIndex="0"
  translateWithId={[Function]}
>
  <div
    className="bx--toolbar-action bx--toolbar-search-container-expandable"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    tabIndex="0"
  >
    <Search
      className="custom-class"
      closeButtonLabelText="Clear search input"
      id="custom-id"
      labelText="Filter table"
      onChange={[Function]}
      placeHolderText="Filter table"
      size="sm"
      tabIndex="-1"
      type="text"
      value=""
    >
      <div
        aria-labelledby="custom-id-search"
        className="bx--search bx--search--sm custom-class"
        role="search"
      >
        <ForwardRef(Search16)
          className="bx--search-magnifier"
        >
          <Icon
            className="bx--search-magnifier"
            fill="currentColor"
            height={16}
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <svg
              aria-hidden={true}
              className="bx--search-magnifier"
              fill="currentColor"
              focusable="false"
              height={16}
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width={16}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15,14.3L10.7,10c1.9-2.3,1.6-5.8-0.7-7.7S4.2,0.7,2.3,3S0.7,8.8,3,10.7c2,1.7,5,1.7,7,0l4.3,4.3L15,14.3z M2,6.5	C2,4,4,2,6.5,2S11,4,11,6.5S9,11,6.5,11S2,9,2,6.5z"
              />
            </svg>
          </Icon>
        </ForwardRef(Search16)>
        <label
          className="bx--label"
          htmlFor="custom-id"
          id="custom-id-search"
        >
          Filter table
        </label>
        <input
          autoComplete="off"
          className="bx--search-input"
          id="custom-id"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Filter table"
          role="searchbox"
          tabIndex="-1"
          type="text"
          value=""
        />
        <button
          aria-label="Clear search input"
          className="bx--search-close bx--search-close--hidden"
          onClick={[Function]}
          type="button"
        >
          <ForwardRef(Close16)>
            <Icon
              fill="currentColor"
              height={16}
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width={16}
              xmlns="http://www.w3.org/2000/svg"
            >
              <svg
                aria-hidden={true}
                fill="currentColor"
                focusable="false"
                height={16}
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
                />
              </svg>
            </Icon>
          </ForwardRef(Close16)>
        </button>
      </div>
    </Search>
  </div>
</TableToolbarSearch>
`;
